graph/telemetry/istio/appender/security_policy.go

```diff
@@ -90,6 +90,12 @@ func (a SecurityPolicyAppender) appendGraph(trafficMap graph.TrafficMap, namespa
 	a.populateSecurityPolicyMap(securityPolicyMap, &inVector)
 
 	applySecurityPolicy(trafficMap, securityPolicyMap)
+
+	principalMap := make(map[string]map[graph.MetadataKey]string)
+	a.populatePrincipalMap(principalMap, &outVector)
+	a.populatePrincipalMap(principalMap, &inVector)
```

Comment:
Not sure you really need to do all of this separately, I'd suggest maybe just working this into the existing 'populate' and 'apply' and not have to walk through everything twice.  If you are worried about getting the feature in then this is fine for now and I'll probably refactor it later.